baseline-error-prone/src/test/java/com/palantir/baseline/errorprone/BadAssertTest.java

```diff
@@ -0,0 +1,112 @@
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.baseline.errorprone;
+
+import org.junit.jupiter.api.Test;
+
+final class BadAssertTest {
+
+    @Test
+    void testNoDescription() {
+        fix().addInputLines(
+                        "Test.java",
+                        // format-hint
+                        "public class Test {",
+                        "  void f(boolean in) {",
+                        "    assert in;",
+                        "  }",
+                        "}")
+                .addOutputLines(
+                        "Test.java",
+                        "public class Test {",
+                        "  void f(boolean in) {",
+                        "    if (!in) {",
+                        "        throw new IllegalStateException();",
+                        "    }",
+                        "  }",
+                        "}")
+                .doTest();
+    }
+
+    @Test
+    void testConstantStringDescription() {
+        fix().addInputLines(
+                        "Test.java",
+                        "public class Test {",
+                        "  void f(boolean in) {",
+                        "    assert in : \"oops\";",
+                        "  }",
+                        "}")
+                .addOutputLines(
+                        "Test.java",
+                        "import com.palantir.logsafe.Preconditions;",
+                        "public class Test {",
+                        "  void f(boolean in) {",
+                        "    Preconditions.checkState(in, \"oops\");",
+                        "  }",
+                        "}")
+                .doTest();
+    }
+
+    @Test
+    void testConstantNonStringDescription() {
+        fix().addInputLines(
+                        "Test.java",
+                        // format-hint
+                        "public class Test {",
+                        "  void f(boolean in) {",
+                        "    assert in : 1;",
+                        "  }",
+                        "}")
+                .addOutputLines(
+                        "Test.java",
+                        "public class Test {",
+                        "  void f(boolean in) {",
+                        "    if (!in) {",
+                        "        throw new IllegalStateException(String.valueOf(1));",
+                        "    }",
+                        "  }",
+                        "}")
+                .doTest();
+    }
+
+    @Test
+    void testNonConstantStringDescription() {
+        fix().addInputLines(
+                        "Test.java",
+                        // format-hint
```

Comment:
formatter puts all these onto one line without a well-placed comment. Some day we'll have text blocks.